dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyLister.java

```diff
@@ -0,0 +1,139 @@
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.collection.DependencyCollectionException;
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.graph.DependencyNode;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.DependencyRequest;
+import org.eclipse.aether.resolution.DependencyResolutionException;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.file.FileTransporterFactory;
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.Files;
+
+/**
+ * Based on <a href="https://maven.apache.org/resolver/index.html">Apache Maven Artifact Resolver</a>
+ * (formerly known as Eclipse Aether).
+ */
+public class DependencyLister {
+  
+  private static final RepositorySystem SYSTEM = newRepositorySystem();
+  private static final RemoteRepository CENTRAL =
+      new RemoteRepository.Builder("central", "default", "http://repo1.maven.org/maven2/").build();
+
+  public static void main(String[] args)
+      throws DependencyCollectionException, DependencyResolutionException {
+    
+    if (args.length != 1 || !args[0].contains(":")) {
+      System.err.println("Usage: java " + DependencyLister.class.getCanonicalName()
+          + " groupdId:artifactId:version");
+      return;
+    }
+    
+    String[] coordinates = args[0].split(":");
+    String groupId = coordinates[0];
+    String artifactId = coordinates[1];
+    String version = coordinates[2];
+    
+    DependencyNode node = resolveCompileTimeDependencies(groupId, artifactId, version);
+    
+    for (DependencyNode dependency : node.getChildren()) {
+      System.out.println(dependency.toString());
+    }
+  }
+
+  private static DependencyNode resolveCompileTimeDependencies(
+      String groupId, String artifactId, String version)
+      throws DependencyCollectionException, DependencyResolutionException {
+    
+    RepositorySystemSession session = newSession();
+
+    Artifact artifact = new DefaultArtifact(groupId + ':' + artifactId + ':' + version);
+    Dependency dependency = new Dependency(artifact, "compile");
+
+    CollectRequest collectRequest = new CollectRequest();
+    collectRequest.setRoot(dependency);
+    collectRequest.addRepository(CENTRAL);
+    DependencyNode node = SYSTEM.collectDependencies(session, collectRequest).getRoot();
+
+    DependencyRequest dependencyRequest = new DependencyRequest();
+    dependencyRequest.setRoot(node);
+
+    SYSTEM.resolveDependencies(session, dependencyRequest);
```

Comment:
It feels wrong to me for an identifier with all-caps to be mutated - this doesn't really feel like a constant... (at least, that's how I read this line) - I'd recommend `system` instead for the name.

Reply:
done, though loooking at it now I'm not sure it is mutated, and I'm not sure it isn't. The dependencyRequest argument is mutated for sure.